### PR TITLE
Add target score validation.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: exam-processor/src/test/resources/ICA-score-test.xml [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+        <Accommodation code="TDS_RSL_ListView" type="Review Screen Layout" value="List" segment="0"/>
+        <Accommodation code="TDS_SCNotepad" type="Student Comments" value="On" segment="0"/>
+        <Accommodation code="TDS_ST1" type="Strikethrough" value="True" segment="0"/>
+        <Accommodation code="TDS_SVC1" type="System Volume Control" value="Show widget" segment="0"/>
+        <Accommodation code="TDS_TPI_ResponsesFix" type="Test Progress Indicator" value="Show indicator as a fraction and adjust to test length" segment="0"/>
+        <Accommodation code="TDS_TS_Universal" type="Test Shell" value="Universal Shell" segment="0"/>
+        <Accommodation code="TDS_WL_Glossary" type="Word List" value="English Glossary" segment="0"/>
+        <Score measureOf="Overall" measureLabel="Attempted" value="Y" standardError=""/>
+        <Score measureOf="Overall" measureLabel="ThetaScore" value="-0.0933834960601748" standardError="0.256414973091891"/>
+        <Score measureOf="10" measureLabel="ThetaScore" value="-0.208050186212807" standardError="0.344876334603102"/>
```
Comment (agorina) [inline exam-processor/src/test/resources/ICA-score-test.xml]: This score is _invalid_, but we are ignoring it.

File: exam-processor/src/test/resources/ICA-score-test.xml [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+        <Score measureOf="SOCK_2" measureLabel="ThetaScore" value="0.118177841870205" standardError="0.43644762387902"/>
+        <Score measureOf="3" measureLabel="ThetaScore" value="-0.184827266208388" standardError="0.699264951337106"/>
+        <Score measureOf="Overall" measureLabel="ScaleScore" value="2507.49468876243" standardError="20.3337073661869"/>
+        <Score measureOf="1" measureLabel="ScaleScore" value="2498.40162023332" standardError="27.348693334026"/>
+        <Score measureOf="SOCK_2" measureLabel="ScaleScore" value="2524.27150286031" standardError="34.6102965736062"/>
+        <Score measureOf="3" measureLabel="ScaleScore" value="2500.24319778967" standardError="55.4517106410325"/>
+        <Score measureOf="Other" measureLabel="PerformanceLevel" value="3" standardError=""/>
```
Comment (agorina) [inline exam-processor/src/test/resources/ICA-score-test.xml]: Ditto.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/Assessment.java [hunk 3]
```diff
@@ -39,7 +45,6 @@ public String getNaturalId() {
         return naturalId;
     }
 
-
```
Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/Assessment.java]: One more just for @malaffoon .

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/repository/impl/JdbcAssessmentRepository.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
     private final NamedParameterJdbcTemplate jdbcTemplate;
 
     @Value("${sql.assessment.findOneByNaturalId}")
-    private String slqFindOneByNaturalId;
+    private String sqlFindOneByNaturalId;
```
Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/repository/impl/JdbcAssessmentRepository.java]: And this one is just for @jtreuting :)

Comment (malaffoon) [reply exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/repository/impl/JdbcAssessmentRepository.java]: See @jtreuting, i told you not to say those mean things about her.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java [hunk 5]
```diff
@@ -222,6 +220,8 @@ private Integer toValidCompleteness(final String value) {
                 } else if (StandardMetRelativeResidualScoreLabel.equals(label)) {
                     builder.standardMetRelativeResidualScore(parserHelper.validate(scoreOf + " " + label, score.getValue(), toResidualScore));
                 }
+            } else if (scoreLabelsToValidate.contains(label)) {
+                errorCollector.add(new DataElementError("scoreOf [" + scoreOf + "] for label [" + label + "]", score.getValue(), "unrecognizable score for the asmt"));
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java]: I'm clearly missing something. If you see the score labels you add an error, no matter what?

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/AssessmentRepositoryIT.java [hunk 6]
```diff
@@ -30,10 +30,14 @@
...
+            "INSERT INTO target (id, claim_id, natural_id) VALUES (-89, -99, 'T1|1-2');",
+            "INSERT INTO target (id, claim_id, natural_id) VALUES (-88, -99, 'T2|A');",
+            "INSERT INTO target (id, claim_id, natural_id) VALUES (-87, -98, 'T3|R-B');",
             " INSERT INTO item (claim_id, target_id, natural_id, asmt_id, dok_id, difficulty, max_points, math_practice, allow_calc) VALUES \n" +
-                    " (1, 1, 'item1', -22, 1, 0.5, 5, null, null)," +
-                    " (1, 1, 'item2', -22, 1, 0.5, 5, null, null)," +
-                    " (1, 1, 'item3', -22, 1, 0.5, 5, null, null);"
+                    " (-99, -89, 'item1', -22, 1, 0.5, 5, null, null)," +
+                    " (-99, -88, 'item2', -22, 1, 0.5, 5, null, null)," +
+                    " (-98, -87, 'item3', -22, 1, 0.5, 5, null, null);"
```
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/AssessmentRepositoryIT.java]: nit: i could use all those extra spaces (after the quote) in the footnotes for my book.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java (hunk 5 above)
Comment (agorina) [reply exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultStudentExamProcessor.java]: This _else_ gets executed only if the score has not been processed ( meaning it is not a claim, overall or asmt's target score). 

Initially I wanted to report an error for any mismatched scores, but then decided that it is going to be too restrictive. 

So if the score has not been identified as a valid score, and it has a label of 'StudentRelativeResidualScoreLabel' or 'StandardMetRelativeResidualScoreLabel' (see line 61), then I report an error.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/AssessmentRepositoryIT.java (hunk 6 above)
Comment (agorina) [reply exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/AssessmentRepositoryIT.java]: Fixing...but now you owe me two chapters in your book.

**Comment by agorina** [comment]
Merging but will address comments if needed.

**Pull request merged**